nilearn/plotting/displays.py

```diff
@@ -266,9 +295,23 @@ def draw_position(self, size, bg_color, **kwargs):
     def _add_markers(self, marker_coords, marker_color, marker_size, **kwargs):
         """Plot markers"""
         marker_coords_2d = _coords_3d_to_2d(marker_coords, self.direction)
-
         xdata, ydata = marker_coords_2d.T
 
+        # Allow markers only in their respective hemisphere when appropriate
+        if self.direction in 'lr':
+            relevant_coords = []
+            xcoords, ycoords, zcoords = marker_coords.T
+            for cidx, xc in enumerate(xcoords):
+                if self.direction == 'r':
+                    if xc >= 0:
+                        relevant_coords.append(cidx)
+                elif self.direction == 'l':
+                    if xc < 0:
```

Comment:
> By the way just wondering what should we do if xc == 0 (maybe that never
> happens in practice I am not sure). Should we not show the marker on both 'l'
> and 'r' plots?

Actually, that's probably a good idea. And that way we have edges going
to it in both plot (this needs to be documented in the docstrings of
add_marker and plot_connectome).